### DISCUSSION TITLE
Implement View Transition Types of the View Transitions API

### DISCUSSION THREAD

**Discussion by phal0r** [remix-run/react-router#12233] [Proposals]
Currently there is basic support for view transitions, which can be enabled using `enableViewTransition`.

Unfortunately quite basic requirements like handling "back" and "forward" directions cannot be achieved with this approach. [This great article](https://developer.chrome.com/docs/web-platform/view-transitions/same-document#view-transition-types) describes different scenarios and the approach to handle direction (and other context one may need) using view transition types.

I think the changes would be quite minimal. A new option viewTransitionTypes needs to be added and the call to document.startViewTransition needs to check for the presence of this option and adjust the structure of the arguments accordingly. Browser support seems to be very good aswell.

**Comment by phal0r** [comment]
Also mentioned here: #11555

**Comment by akrynski-1** [comment]
Not having the ability to distinguish between forwards and backwards navigation reduces amount of possible animations that can be build, for example slide.
It would be nice to have a way to do that.
